vector/src/androidTest/java/im/vector/app/core/utils/WaitUntil.kt

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright (c) 2022 New Vector Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package im.vector.app.core.utils
+
+import kotlinx.coroutines.delay
+import org.amshove.kluent.fail
+import kotlin.time.Duration
+import kotlin.time.Duration.Companion.milliseconds
+import kotlin.time.Duration.Companion.seconds
+
+/**
+ * Tries a [condition] several times until it returns true or a [timeout] is reached waiting for some [delay] time between retries.
+ * On timeout it fails with an [errorMessage].
+ */
+suspend fun waitUntilCondition(
+        errorMessage: String,
+        timeout: Duration = 1.seconds,
+        delay: Duration = 50.milliseconds,
```

Comment:
tiny nit, maybe just me :smile:  on first read `delay` makes me think this is _only_ an initial delay, `interval` or `retryDelay` might give a little extra context~